token/client/src/token.rs

```diff
@@ -3846,4 +3847,77 @@ where
         )
         .await
     }
+
+    /// Initialize token-group on a mint
+    pub async fn token_group_initialize<S: Signers>(
+        &self,
+        mint_authority: &Pubkey,
+        update_authority: &Pubkey,
+        max_size: u32,
+        signing_keypairs: &S,
+    ) -> TokenResult<T::Output> {
+        self.process_ixs(
+            &[spl_token_group_interface::instruction::initialize_group(
+                &self.program_id,
+                &self.pubkey,
+                &self.pubkey,
+                mint_authority,
+                Some(*update_authority),
+                max_size,
+            )],
+            signing_keypairs,
+        )
+        .await
+    }
+
+    async fn get_additional_rent_for_fixed_len_extension<V: Extension + Pod>(
+        &self,
+    ) -> TokenResult<u64> {
+        let account = self.get_account(self.pubkey).await?;
+        let account_lamports = account.lamports;
+        let new_account_len = account
+            .data
+            .len()
+            .saturating_add(size_of::<ExtensionType>())
+            .saturating_add(size_of::<Length>())
+            .saturating_add(size_of::<V>());
+        let new_rent_exempt_minimum = self
+            .client
+            .get_minimum_balance_for_rent_exemption(new_account_len)
+            .await
+            .map_err(TokenError::Client)?;
+        Ok(new_rent_exempt_minimum.saturating_sub(account_lamports))
+    }
```

Comment:
Since you're already fetching the account here, you can do a quick to see if the account already has the extension, and return 0 if it does.

Reply:
Yep totally, I just wasn't sure how useful it would be since it's basically like saying "hey you don't need any rent because the extension exists, but the program is going to barf on you when this goes through".

Reply:
But this call will return the program's error so I guess the 0 approach is fine.